src/common/errors.ts

```diff
@@ -51,6 +51,45 @@ ${raw ? `Response: ${JSON.stringify(raw, null, 2)}` : ""}
   }
 }
 
+type ErrorBody = {
+  code?: string;
+  name?: string;
+  severity?: string;
+  source?: string;
+  description?: string;
+  resolution?: string;
+  helpLink?: string;
+  location?: {
+    failingLine?: number;
+    startOffset?: number;
+    endOffset?: number;
+  };
+};
+
+export class CompositeError extends Error {
+  message: string;
+  errorsArray: Array<ErrorBody>;
+
+  constructor(errors: Array<ErrorBody>) {
+    const parsedErrors = errors.map(error => {
+      const severity = error.severity ? `${error.severity}: ` : "";
+      const name = error.name ? `${error.name} ` : "";
+      const code = error.code ? `(${error.code}) ` : "";
+      const description = error.description ? `- ${error.description}` : "";
+      const helpLink = error.helpLink ? `, see ${error.helpLink}` : "";
+      const location = error.location
+        ? ` at ${JSON.stringify(error.location)}`
+        : "";
+      // "{severity}: {name} ({code}) - {description} at {location} see {helpLink}"
+      return `${severity}${name}${code}${description}${location}${helpLink}`;
+    });
+    const formattedMessage = `${parsedErrors.join(",\n")}`;
```

Comment:
IMHO if `errors` exists but empty we should generarate some kind of default error message, e.g. "Unknown error"

Reply:
I see where you're coming from, but this is the response that's returned with a 200/201 code. It becomes somewhat ambiguous in this case. A query succeeded and errors are empty therefore the query is ok? Possibly. To avoid confusion I'm following the spec in FIR-33624